f2/src/utils/getData.js

```diff
@@ -51,7 +51,12 @@ async function getData(data, files) {
   data.evidence.files = filesToJson
 
   const selfHarmWords = selfHarmWordsScan(data)
-  if (selfHarmWords) {
+
+  if (
+    selfHarmWords !== null &&
```

Comment:
Why are we converting an array to String here? I think a better approach might be something like this:

`selfHarmWords === undefined || selfHarmWords.length == 0`

If it does not exist or if there are no elements in the array it will return true.

Reply:
Because there is still possibility of one element added to array, and this element contains empty value or null value

Reply:
If that is a possibility it is a bug. We should address it in this PR.

Reply:
Bug not in this specific case, but I saw similar situation many times in other projects, so I take this much stricter condition to rule out such things.

Reply:
This condition is only used to write a message to the console. The array will be used in other parts of the app, like when creating the analyst email. If this function will return an array containing blank strings or null values it should be fixed. If not we shouldn't have this check.

This check doesn't consider JavaScript types. If something goes wrong the array will be `undefined` instead of `null`. Calling toString on `undefined `will cause a `TypeError`. From Mozilla:

"A variable that has not been assigned a value is of type undefined. A method or statement also returns undefined if the variable that is being evaluated does not have an assigned value. A function returns undefined if a value was not returned."

[https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/undefined](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/undefined)

Reply:
Null is removed in latest commit.

Reply:
String length check is also removed to get it pass review. while I still think, as a preventive measure, it is useful to have such check to prevent such possible scenario.